ipld/get_shares.go

```diff
@@ -99,7 +113,14 @@ func GetShares(ctx context.Context, bGetter blockservice.BlockGetter, root cid.C
 						// calc position for children nodes (bin-tree-feat),
 						// s.t. 'if' above knows where to put a share
 						pos: j.pos*2 + i,
+						// we pass the context to job so that spans are tracked in a tree
+						// structure
+						ctx: ctx,
 					}:
+						span.AddEvent("added-job", trace.WithAttributes(
+							attribute.String("cid", lnk.Cid.String()),
+							attribute.Int("pos", j.pos*2+i),
+						))
```

Comment:
Pls, add those attributes to the `process-job` span instead of event, as mentioned in the last review

Reply:
Similarly for the namespace pls

Reply:
Then you won't need to record error with attributes on loc 170 as well

Reply:
because the error is attached to the span which would have those attributes attached there already

Reply:
Ohh now I get it. Makes sense, also removing the added-job event in total because the new correct tree structure makes it obselete